cmd/keeper/keeper.go

```diff
@@ -1039,18 +1039,35 @@ func (p *PostgresKeeper) resyncIfNotReady(followed *cluster.KeeperState, initial
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
 	if !ready {
-		log.Info("Standby is not accepting connections. It's probably waiting for unavailable WALs. Forcing a full resync.")
-		if err = p.resync(followed, initialized, started); err != nil {
-			return trace.Wrap(err, "failed to full resync from followed instance")
+		log.Info("Standby is not accepting connections. Forcing a full resync.")
+		if err = p.resyncAndStart(followed, initialized, started); err != nil {
+			return trace.Wrap(err)
 		}
-		if err = p.pgm.Start(); err != nil {
-			return trace.Wrap(err, "error starting PostgreSQL instance")
+	}
+
+	if err = p.pgm.IsStreaming(); err != nil {
```

Comment:
Seems like this is going to `resyncAndStart` if either `!ready` or `!IsStreaming` - did you mean to check them and restart sequentially or can the `resyncAndStart` be done once, i.e.:
```go
if !ready || p.pgm.IsStreaming() != nil { // Note - not actual code, but an example to demonstrate the point
	p.resyncAndStart(...)
}
```